### PR TITLE
Documentation: Document @storybook/test package

### PR DESCRIPTION
**Pull request by yannbf**
Relates to #23871

## What I did

This PR proposes changes to the documentation in Storybook 8, following the requirements of #23871, to:
- use @storybook/test instead of @storybook/testing-library and @storybook/jest in every setup docs
- use @storybook/test instead of @storybook/testing-library and @storybook/jest in every code snippet
- de-emphasize argTypesRegex
- revamp Addon-actions docs to mention about the explicit `fn` functions and its connection to actions panel

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/essentials/actions.md
Comment (kylegach) [inline docs/essentials/actions.md:22]:
```suggestion
The recommended way to write actions is to use the `fn` utility from `@storybook/test` to mock and spy args. This is very useful for writing [interaction tests](../writing-tests/interaction-testing.md). You can mock your component's methods by assigning them to the `fn()` function:
```

File: docs/essentials/actions.md
Comment (kylegach) [inline docs/essentials/actions.md:36]:
```suggestion
If your component calls an arg (because of either the user's interaction or the `play` function) and that arg is spied on , the event will show up in the action panel:
```

**Comment by kylegach** [comment]
Noting for posterity:

Once this PR is merged, two things need to happen, in this order (ping me to do it):

1. Configure frontpage's Netlify to build the `release-8-0` branch
2. Merge https://github.com/storybookjs/frontpage/pull/603

**Comment by yannbf** [comment]
Like we discussed yesterday, there might be the need of writing a migration note, and checking whether the new package clashes with the other packages (sb/testing-lib, sb/jest)

@kasperpeulen can you do that?

**Comment by kasperpeulen** [comment]
@kylegach I would like to merge this PR and get it in the docs, if you are okay with that :)

**Review by kasperpeulen — APPROVED** [review]
LGTM!

**Comment by yannbf** [comment]
> @yannbf I found that the packages don't clash anymore, and can be used together in stories, when migrating. Shall we make the migration note part of this PR?
> 
> I think it would be good already start documenting the test package also in the README of the package as @IanVS suggested.

Yes please, feel free to do that!

**Pull request merged**
